nexus/src/app/background/blueprint_load.rs

```diff
@@ -0,0 +1,176 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Background task for loading the target blueprint from the DB
+//!
+//! This task triggers the `blueprint_execution` background task when the
+//! blueprint changes.
+
+use super::common::BackgroundTask;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db::DataStore;
+use nexus_types::deployment::Blueprint;
+use serde_json::json;
+use std::sync::Arc;
+use tokio::sync::watch;
+
+pub struct TargetBlueprintLoader {
+    datastore: Arc<DataStore>,
+    last: Option<Arc<Blueprint>>,
+    tx: watch::Sender<Option<Arc<Blueprint>>>,
+    rx: watch::Receiver<Option<Arc<Blueprint>>>,
+}
+
+impl TargetBlueprintLoader {
+    pub fn new(datastore: Arc<DataStore>) -> TargetBlueprintLoader {
+        let (tx, rx) = watch::channel(None);
+        TargetBlueprintLoader { datastore, last: None, tx, rx }
+    }
+
+    /// Expose the target blueprint
+    pub fn watcher(&self) -> watch::Receiver<Option<Arc<Blueprint>>> {
+        self.rx.clone()
```

Comment:
Tiny nit - I think we could use `self.tx.subscribe()` here and drop `self.rx` entirely (unless somewhere else we're assuming there's always at least one receiver, which I'm not seeing at a glance)

Reply:
Good call! Fixed in 769cda3c5c3a5209c40b4f635ac4bcfd7fefbf47